### PR TITLE
program for calculating nCr added in python language

### PR DESCRIPTION
**Pull request by dhruvdy**

### CODE REVIEW

**Comment by dhruvdy** [comment]
Issue #664

File: Algorithm/Number_theory/readme.md [hunk 1]
```diff
@@ -43,5 +43,6 @@
 - Determining if a number is a prime number or not using Trial Division Algorithm ---> [C++](/Code/C++/prime_number.cpp)
 - Factorial of a number without recursion ---->[Python](/Code/Python/factorial_without_recursion.py)
```
Comment (rish-singhal) [inline Algorithm/Number_theory/readme.md:44]: sort this list in alphabetical order

File: Code/Python/calculate_the_value_of_nCr.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#we divide both numerator and denominator by gcd() numbers as to make calculation simple 
+#and to ensue the result may not go out of range
+
+
+def gcd(num1,num2):             #This function is for calculating greatest common divisor
+    if num2==0:
+        return num1
+    return gcd(num2,num1%num2)
+
+
```
Comment (rish-singhal) [inline Code/Python/calculate_the_value_of_nCr.py:13]: remove extra lines

File: Code/Python/calculate_the_value_of_nCr.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+#This is the program to calcuate the nCr.
+
+#In this method used for finding nCr ,while calculating nCr   n(n-1)(n-2)---(n-r+1)/r(r-1)---1
+#we divide both numerator and denominator by gcd() numbers as to make calculation simple 
+#and to ensue the result may not go out of range
+
+
+def gcd(num1,num2):             #This function is for calculating greatest common divisor
```
Comment (rish-singhal) [inline Code/Python/calculate_the_value_of_nCr.py:8]: dont give this much space between comment and def, rather use docstring for the function i.e.

```python
def func():
    """ XYZ.... """
```

do this for all the functions in this file

File: Code/Python/calculate_the_value_of_nCr.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    if num2==0:
```
Comment (rish-singhal) [inline Code/Python/calculate_the_value_of_nCr.py:9]: give spaces around "," "=",.... and all

File: Code/Python/calculate_the_value_of_nCr.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        return n
+    
+    n_limit=n-r+1
+    numerator=1     
+    denominator=1
+    #while loop will perform the calculation for nCr n(n-1)(n-2)---(n-r+1)/r(r-1)(r-2)--1
+    while n>=n_limit and r>=1:   
+        numerator*=n
+        denominator*=r
+        divisor=gcd(numerator,denominator)
```
Comment (rish-singhal) [inline Code/Python/calculate_the_value_of_nCr.py:33]: here to give spaces

File: Code/Python/calculate_the_value_of_nCr.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+Output:
+184756
+
+Test Case: 2
+Enter the value of n in nCr 25
+Enter the value of n in nCr 12
+3268760
+    
+Time complexity : O(rlog(n))
```
Comment (rish-singhal) [inline Code/Python/calculate_the_value_of_nCr.py:63]: what is r and n, describe here

File: Code/Python/calculate_the_value_of_nCr.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        n=n-1
+        r=r-1
+        
+    return int(numerator/denominator) 
+
+
+n=input("Enter the value of n in nCr ")
+r=input("Enter the value of r in nCr ")
+n=int(n)
+r=int(r)
```
Comment (rish-singhal) [inline Code/Python/calculate_the_value_of_nCr.py:45]: give spaces

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by rish-singhal** [comment]
solve merge conflicts

**Pull request closed without merging**